diff -ur kernel/msm-3.18.orig/fs/ecryptfs/mmap.c kernel/msm-3.18/fs/ecryptfs/mmap.c
--- kernel/msm-3.18.orig/fs/ecryptfs/mmap.c	2016-09-17 01:59:49.000000000 -0700
+++ kernel/msm-3.18/fs/ecryptfs/mmap.c	2017-01-08 11:57:32.755169410 -0800
@@ -512,11 +512,12 @@
 	rc = lower_inode->i_op->setxattr(lower_dentry, ECRYPTFS_XATTR_NAME,
 					 xattr_virt, size, 0);
 	mutex_unlock(&lower_inode->i_mutex);
-	if (rc)
+	if (rc) {
 		printk(KERN_ERR "Error whilst attempting to write inode size "
 		       "to lower file xattr; rc = [%d]\n", rc);
 		printk(KERN_ERR " [CCAudit] Error whilst attempting to write inode size "
 		       "to lower file xattr; rc = [%d]\n", rc);
+	}
 	kmem_cache_free(ecryptfs_xattr_cache, xattr_virt);
 out:
 	return rc;
