diff -ur kernel/msm-3.18.orig/drivers/regulator/cpr3-regulator.c kernel/msm-3.18/drivers/regulator/cpr3-regulator.c
--- kernel/msm-3.18.orig/drivers/regulator/cpr3-regulator.c	2016-09-17 01:59:48.000000000 -0700
+++ kernel/msm-3.18/drivers/regulator/cpr3-regulator.c	2017-01-08 11:57:32.755169410 -0800
@@ -6088,7 +6088,7 @@
  */
 int cpr3_regulator_unregister(struct cpr3_controller *ctrl)
 {
-	int i, j, rc = 0;
+	int i, j, rc;
 
 	mutex_lock(&cpr3_controller_list_mutex);
 	list_del(&ctrl->list);
@@ -6098,11 +6098,12 @@
 	if (ctrl->irq && !cpumask_empty(&ctrl->irq_affinity_mask))
 		unregister_hotcpu_notifier(&ctrl->cpu_hotplug_notifier);
 
-	if (ctrl->ctrl_type == CPR_CTRL_TYPE_CPR4)
+	if (ctrl->ctrl_type == CPR_CTRL_TYPE_CPR4) {
 		rc = cpr3_ctrl_clear_cpr4_config(ctrl);
 		if (rc)
 			cpr3_err(ctrl, "failed to clear CPR4 configuration,rc=%d\n",
 				rc);
+	}
 
 	cpr3_ctrl_loop_disable(ctrl);
 
