diff -ur kernel/msm-3.18.orig/drivers/media/platform/msm/camera_v2/isp/msm_isp44.c kernel/msm-3.18/drivers/media/platform/msm/camera_v2/isp/msm_isp44.c
--- kernel/msm-3.18.orig/drivers/media/platform/msm/camera_v2/isp/msm_isp44.c	2016-09-17 01:59:46.000000000 -0700
+++ kernel/msm-3.18/drivers/media/platform/msm/camera_v2/isp/msm_isp44.c	2017-01-08 11:57:32.751169393 -0800
@@ -507,8 +507,9 @@
 			msm_isp_notify(vfe_dev, ISP_EVENT_SOF, VFE_PIX_0, ts);
 			if (vfe_dev->axi_data.stream_update[VFE_PIX_0])
 				msm_isp_axi_stream_update(vfe_dev, VFE_PIX_0);
-				vfe_dev->hw_info->vfe_ops.core_ops.reg_update(
-				   vfe_dev, VFE_PIX_0);
+/* BUG ALERT!  Was the following line supposed to be guarded by the if() ??? */
+			vfe_dev->hw_info->vfe_ops.core_ops.reg_update(
+				vfe_dev, VFE_PIX_0);
 		}
 	}
 }
diff -ur kernel/msm-3.18.orig/drivers/media/platform/msm/vidc/msm_venc.c kernel/msm-3.18/drivers/media/platform/msm/vidc/msm_venc.c
--- kernel/msm-3.18.orig/drivers/media/platform/msm/vidc/msm_venc.c	2016-09-17 01:59:46.000000000 -0700
+++ kernel/msm-3.18/drivers/media/platform/msm/vidc/msm_venc.c	2017-01-08 11:57:32.751169393 -0800
@@ -1576,17 +1576,18 @@
 			V4L2_CID_MPEG_VIDC_VIDEO_EXTRADATA);
 		if (ctrl)
 			extradata = v4l2_ctrl_g_ctrl(ctrl);
-			switch (extradata) {
-			case V4L2_MPEG_VIDC_EXTRADATA_INPUT_CROP:
-			case V4L2_MPEG_VIDC_EXTRADATA_DIGITAL_ZOOM:
-			case V4L2_MPEG_VIDC_EXTRADATA_ASPECT_RATIO:
-			case V4L2_MPEG_VIDC_EXTRADATA_YUV_STATS:
-			case V4L2_MPEG_VIDC_EXTRADATA_ROI_QP:
-				*num_planes = *num_planes + 1;
-				break;
-			default:
-				break;
-			}
+/*FIXME: this is worrisome, was extradata deliberately left uncleared above?*/
+		switch (extradata) {
+		case V4L2_MPEG_VIDC_EXTRADATA_INPUT_CROP:
+		case V4L2_MPEG_VIDC_EXTRADATA_DIGITAL_ZOOM:
+		case V4L2_MPEG_VIDC_EXTRADATA_ASPECT_RATIO:
+		case V4L2_MPEG_VIDC_EXTRADATA_YUV_STATS:
+		case V4L2_MPEG_VIDC_EXTRADATA_ROI_QP:
+			*num_planes = *num_planes + 1;
+			break;
+		default:
+			break;
+		}
 
 		inst->fmts[OUTPUT_PORT]->num_planes = *num_planes;
 		rc = call_hfi_op(hdev, session_set_property, inst->session,
diff -ur kernel/msm-3.18.orig/drivers/misc/hdcp.c kernel/msm-3.18/drivers/misc/hdcp.c
--- kernel/msm-3.18.orig/drivers/misc/hdcp.c	2016-09-17 01:59:47.000000000 -0700
+++ kernel/msm-3.18/drivers/misc/hdcp.c	2017-01-08 11:57:32.751169393 -0800
@@ -1295,11 +1295,12 @@
 	/* check if it's a repeater */
 	if ((rsp_buf->msg[0] == SKE_SEND_EKS_MESSAGE_ID) &&
 			(rsp_buf->msglen == SKE_SEND_EKS_MESSAGE_SIZE)) {
+/*FIXME: worrisome, are these really supposed to be split, compiler warning...*/
 		if ((rsp_buf->flag ==
 			HDCP_TXMTR_SUBSTATE_WAITING_FOR_RECIEVERID_LIST) &&
 						(rsp_buf->timeout > 0))
 			handle->repeater_flag = true;
-			handle->update_stream = true;
+		handle->update_stream = true;
 	}
 
 	memset(handle->listener_buf, 0, MAX_TX_MESSAGE_SIZE);
