diff -ur kernel/msm-3.18.orig/sound/soc/msm/apq8096-auto.c kernel/msm-3.18/sound/soc/msm/apq8096-auto.c
--- kernel/msm-3.18.orig/sound/soc/msm/apq8096-auto.c	2016-09-17 01:59:50.000000000 -0700
+++ kernel/msm-3.18/sound/soc/msm/apq8096-auto.c	2017-01-08 18:08:29.509319418 -0800
@@ -32,7 +32,7 @@
 #include <sound/q6core.h>
 #include <sound/pcm_params.h>
 #include <sound/info.h>
-#include <device_event.h>
+#include "device_event.h"
 #include "qdsp6v2/msm-pcm-routing-v2.h"
 
 #define DRV_NAME "apq8096-auto-asoc-snd"
diff -ur kernel/msm-3.18.orig/sound/soc/msm/apq8096-i2c.c kernel/msm-3.18/sound/soc/msm/apq8096-i2c.c
--- kernel/msm-3.18.orig/sound/soc/msm/apq8096-i2c.c	2016-09-17 01:59:50.000000000 -0700
+++ kernel/msm-3.18/sound/soc/msm/apq8096-i2c.c	2017-01-08 18:09:28.649536941 -0800
@@ -31,7 +31,7 @@
 #include <sound/q6core.h>
 #include <sound/pcm_params.h>
 #include <sound/info.h>
-#include <device_event.h>
+#include "device_event.h"
 #include "qdsp6v2/msm-pcm-routing-v2.h"
 #include "../codecs/wcd9xxx-common.h"
 #include "../codecs/wcd9330.h"
diff -ur kernel/msm-3.18.orig/sound/soc/msm/msm8996.c kernel/msm-3.18/sound/soc/msm/msm8996.c
--- kernel/msm-3.18.orig/sound/soc/msm/msm8996.c	2016-09-17 01:59:50.000000000 -0700
+++ kernel/msm-3.18/sound/soc/msm/msm8996.c	2017-01-08 18:06:44.716916787 -0800
@@ -31,7 +31,7 @@
 #include <sound/q6core.h>
 #include <sound/pcm_params.h>
 #include <sound/info.h>
-#include <device_event.h>
+#include "device_event.h"
 #include "qdsp6v2/msm-pcm-routing-v2.h"
 #include "../codecs/wcd9xxx-common.h"
 #include "../codecs/wcd9330.h"
