diff -ur kernel/msm-3.18.orig/drivers/usb/gadget/function/f_gsi.c kernel/msm-3.18/drivers/usb/gadget/function/f_gsi.c
--- kernel/msm-3.18.orig/drivers/usb/gadget/function/f_gsi.c	2016-09-17 01:59:48.000000000 -0700
+++ kernel/msm-3.18/drivers/usb/gadget/function/f_gsi.c	2017-01-08 18:01:46.399771486 -0800
@@ -18,7 +18,7 @@
 #include <linux/ipa_usb.h>
 #include "f_gsi.h"
 #include "rndis.h"
-#include "debug.h"
+#include "../debug.h"
 
 static unsigned int gsi_in_aggr_size;
 module_param(gsi_in_aggr_size, uint, S_IRUGO | S_IWUSR);
diff -ur kernel/msm-3.18.orig/drivers/usb/gadget/function/f_midi.c kernel/msm-3.18/drivers/usb/gadget/function/f_midi.c
--- kernel/msm-3.18.orig/drivers/usb/gadget/function/f_midi.c	2016-09-17 01:59:48.000000000 -0700
+++ kernel/msm-3.18/drivers/usb/gadget/function/f_midi.c	2017-01-08 17:56:54.926654339 -0800
@@ -32,7 +32,7 @@
 #include <linux/usb/audio.h>
 #include <linux/usb/midi.h>
 
-#include "u_f.h"
+#include "../u_f.h"
 
 MODULE_AUTHOR("Ben Williamson");
 MODULE_LICENSE("GPL v2");
diff -ur kernel/msm-3.18.orig/drivers/usb/gadget/function/f_mtp.c kernel/msm-3.18/drivers/usb/gadget/function/f_mtp.c
--- kernel/msm-3.18.orig/drivers/usb/gadget/function/f_mtp.c	2016-09-17 01:59:48.000000000 -0700
+++ kernel/msm-3.18/drivers/usb/gadget/function/f_mtp.c	2017-01-08 17:59:26.255234264 -0800
@@ -40,7 +40,7 @@
 #include <linux/configfs.h>
 #include <linux/usb/composite.h>
 
-#include "configfs.h"
+#include "../configfs.h"
 
 #define MTP_RX_BUFFER_INIT_SIZE    1048576
 #define MTP_BULK_BUFFER_SIZE       16384
diff -ur kernel/msm-3.18.orig/drivers/usb/gadget/function/f_rndis.c kernel/msm-3.18/drivers/usb/gadget/function/f_rndis.c
--- kernel/msm-3.18.orig/drivers/usb/gadget/function/f_rndis.c	2016-09-17 01:59:48.000000000 -0700
+++ kernel/msm-3.18/drivers/usb/gadget/function/f_rndis.c	2017-01-08 18:00:33.823493921 -0800
@@ -27,7 +27,7 @@
 #include "u_ether_configfs.h"
 #include "u_rndis.h"
 #include "rndis.h"
-#include "configfs.h"
+#include "../configfs.h"
 
 /*
  * This function is an RNDIS Ethernet port -- a Microsoft protocol that's
